### PR TITLE
add /System/Meta* recursively pretty print

### LINKED ISSUE #3228: keys: better PrettyPrint for meta keys
`keys.PrettyPrint` displays meta keys as: `/System/Meta2/"\xff\f\x03\xe8"`. Meta keys are keys about keys. `PrettyPrint` should recursively pretty print the `\xff\f\x03\xe8` suffix of the meta key which is itself a key.

@veteranlu

### PR DESCRIPTION
**Pull request by veteranlu**
@petermattis 
- fixes #3228
- add /System/Meta1/Max and /System/Meta2/Max

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3237)

### CODE REVIEW

File: keys/printer.go [hunk 1]
```diff
@@ -312,11 +292,72 @@ func PrettyPrint(key roachpb.Key) string {
...
-			return buf.String()
+			return buf.String(), formatted
 		}
 	}
 
-	return fmt.Sprintf("%q", []byte(key))
+	return fmt.Sprintf("%q", []byte(key)), false
+}
+
+// PrettyPrint print the key in a human readable format, which organized as:
```
Comment (petermattis) [inline keys/printer.go]: `s/which organized/which is organized/g`

File: keys/printer.go [hunk 2]
```diff
@@ -248,53 +266,15 @@ func decodeKeyPrint(key roachpb.Key) string {
...
-func PrettyPrint(key roachpb.Key) string {
-	if bytes.Equal(key, MaxKey) {
-		return "/Max"
-	} else if bytes.Equal(key, MinKey) {
-		return "/Min"
-	}
-
+func prettyPrintInternal(key roachpb.Key) (string, bool) {
 	var buf bytes.Buffer
+	formatted := true
```
Comment (petermattis) [inline keys/printer.go]: You're never setting this variable to a different value. I'd just remove it and return `true` in the places it is being used.

File: keys/printer.go [hunk 3]
```diff
@@ -58,6 +66,16 @@ var (
...
 	}
 
+	// keyofKeyDict means the key of suffix which is itself a key,  should recursively pretty print it, see issue #3228
+	keyOfKeyDict = []struct {
+		name   string
+		prefix []byte
+	}{
+		{name: "/System/Meta2", prefix: Meta2Prefix},
+		{name: "/System/Meta1", prefix: Meta1Prefix},
+		//{name: "Meta", prefix: MetaPrefix},
```
Comment (petermattis) [inline keys/printer.go]: I think this line can be deleted.

**Comment by petermattis** [comment]
LGTM

File: keys/printer.go [hunk 4]
```diff
@@ -46,18 +56,25 @@ var (
 			{name: "/Range", prefix: LocalRangePrefix, ppFunc: localRangeKeyPrint},
 		}},
 		{name: "/System", start: SystemPrefix, end: SystemMax, entries: []dictEntry{
-			{name: "/Meta2", prefix: Meta2Prefix, ppFunc: print},
-			{name: "/Meta1", prefix: Meta1Prefix, ppFunc: print},
-			//{name: "Meta", prefix: MetaPrefix},
+
```
Comment (tamird) [inline keys/printer.go]: can you remove this line?

**Comment by tamird** [comment]
LGTM

File: keys/printer.go (hunk 4 above)
Comment (veteranlu) [inline keys/printer.go]: @tamird sure.

**Comment by tamird** [comment]
@veteranlu looks like this needs a rebase

**Comment by tamird** [comment]
@veteranlu can you rebase this?

**Comment by petermattis** [comment]
@veteranlu I'll rebase and fix up any remaining issues on Monday if you haven't re-emerged by then.

**Pull request closed without merging**